libs/community/langchain_community/vectorstores/duckdb.py

```diff
@@ -153,10 +168,19 @@ def add_texts(
                 if metadatas and idx < len(metadatas)
                 else None
             )
-            self._connection.execute(
-                f"INSERT INTO {self._table_name} VALUES (?,?,?,?)",
-                [ids[idx], text, embedding, metadata],
+            data.append(
+                {
+                    self._id_key: ids[idx],
+                    self._text_key: text,
+                    self._vector_key: embedding,
+                    "metadata": metadata,
+                }
             )
+        # noinspection PyUnusedLocal
+        df = pd.DataFrame.from_dict(data)  # noqa: F841
+        self._connection.execute(
+            f"INSERT INTO {self._table_name} SELECT * FROM df",
```

Comment:
Hey, 
I am afraid we need it.
DuckDB does not provide any other way to import data efficiently.
Respectively, they provide import from file(CSV, PARQUET, JSON), but it would require writing the file to disk, which is IMO risky (which folder? privileges? ...).
If the new dependency becomes a blocker, I can implement a file approach, but then I would appreciate guidance on how to do it properly.

Reply:
what if we try imprting pandas and if its available we do the updated approach, and if its not available we fallback to the current approach

Reply:
I tried to do it.
But when I tested it end-to-end, it failed here:
https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/duckdb.py#L192
The similarity_search() method uses `fetchdf()` function which requires Pandas.
So the hard dependency was there before.
 
Do you want me to update the similarity_search() method too?

I noticed that you added a new code raising an exception if the import of Pandas fails.
Can we keep the current code in the PR and merge it?

Reply:
I suggest to keep it as it is. 
If we need to remove the dependency, let's create a new issue and solve it separately.